### DISCUSSION TITLE
[General] <How does zksync Era handle the the consistency between bn256 and bls12_381?>

### DISCUSSION THREAD

**Discussion by xrqin** [zkSync-Community-Hub/zksync-developers#416] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Ethereum only has precompiles for one pairing-friendly elliptic curve, BN254. The EIP4844, however, uses another curve, BLS12-381, for KZG commitments. Given now that zksync has activated the eip 4844, I am curious on how zksync Era ensure the consistency between bn256 and bls12_381? Where can I find the corresponding code?

### What is the specific issue or error you're encountering?

.

### Can you share the error messages or logs you're receiving, if any?

.

### Have you made any recent changes to the contract before encountering this issue?

.

### Are there any external libraries or contracts that your contract interacts with?

.

### Can you provide the relevant portions of your contract code where the issue is occurring?

.

### Have you tried to isolate the problem, and if so, what were the results?

.

### What steps have you already taken to try to resolve the issue?

.

**Comment by AlbionaHoti** [comment, marked as answer]
Hi @xrqin, This is a great question! We do have docs on the zksync-era repo that walk you through the 4844 at zkSync. Check this one here: https://github.com/matter-labs/zksync-era/blob/4fd0205676e49660cc1d338953bffe1a47f20d1c/docs/guides/advanced/data/README.md. Let me know if this answers your question! If not, I'll check with the protocol team!
